### PR TITLE
DEV-1339 Add tests for helper functions

### PR DESCRIPTION
**Pull request by WardMaes**
Add tests for following helper functions:
- `try_to_find_md5`
- `is_event_valid`

### CODE REVIEW

**Comment by RudolfDG** [comment]
It would be cleaner to use pytest instead of the standard unittest library. This would allow to remove the class `TestHelperFunctions` and the `if __name__ ==  "main"`.
The tests would look like this:
```
import json
import pytest
from meemoo.helpers import (
    try_to_find_md5,
    get_from_event,
    normalize_or_id,
    is_event_valid,
    InvalidEventException,
)
from tests.resources import S3_MOCK_ESSENCE_EVENT


def test_try_to_find_md5_valid_x_md5sum_meta():
    # Arrange
    metadata = {'x-md5sum-meta': '1234abcd1234abcd1234abcd1234abcd'}
    # Act
    md5 = try_to_find_md5(metadata)
    # Assert
    assert md5 == metadata['x-md5sum-meta']
```
Which is imo a lot cleaner and expressive.

**Comment by WardMaes** [comment]
> 
> 
> It would be cleaner to use pytest instead of the standard unittest library. This would allow to remove the class `TestHelperFunctions` and the `if __name__ == "main"`.
> The tests would look like this:
> 
> ```
> import json
> import pytest
> from meemoo.helpers import (
>     try_to_find_md5,
>     get_from_event,
>     normalize_or_id,
>     is_event_valid,
>     InvalidEventException,
> )
> from tests.resources import S3_MOCK_ESSENCE_EVENT
> 
> 
> def test_try_to_find_md5_valid_x_md5sum_meta():
>     # Arrange
>     metadata = {'x-md5sum-meta': '1234abcd1234abcd1234abcd1234abcd'}
>     # Act
>     md5 = try_to_find_md5(metadata)
>     # Assert
>     assert md5 == metadata['x-md5sum-meta']
> ```
> 
> Which is imo a lot cleaner and expressive.

I initially followed the style of the existing tests in the file. 
The other test files (`test_main.py`, `test_services.py`) make use of the `pytest` library as well, so it makes sense to update this file accordingly.

**Review by maartends — APPROVED** [review]
Nice! :+1: 

Minor remark: with every `assert "not found or syntactically incorrect" in str(excinfo.value)` we're actually testing the implementation of the `InvalidEventException` as well, which is _strictu sensu_ not part of what is being tested. We'll leave it for now.

File: tests/test_helpers.py [hunk 1]
```diff
@@ -15,90 +15,158 @@
...
+    # Assert
+    assert normalized_or_id == "MM-7p8tc89"
+
+
+def test_normalize_or_id_invalid_noid_length_too_short():
+    # Arrange
+    or_id = "or-a1b2c3"
+    # Act and Assert
+    with raises(ValueError) as excinfo:
         normalized_or_id = normalize_or_id(or_id)
```
Comment (maartends) [inline tests/test_helpers.py]: For example:
```suggestion
        _ = normalize_or_id(or_id)
```

**Review by maartends — CHANGES_REQUESTED** [review]
Nice changes to make the linter happy! However, some linting warnings remain: 5 variables in the tests are assigned but not used. These warning can be remedied by using the underscore throwaway variable:

```diff
diff --git a/tests/test_helpers.py b/tests/test_helpers.py
index 0878d04..15b8e57 100644
--- a/tests/test_helpers.py
+++ b/tests/test_helpers.py
@@ -103,7 +103,7 @@ def test_normalize_or_id_invalid_noid_length_too_short():
     or_id = "or-a1b2c3"
     # Act and Assert
     with raises(ValueError) as excinfo:
-        normalized_or_id = normalize_or_id(or_id)
+        _ = normalize_or_id(or_id)
     assert "Invalid noid length" in str(excinfo.value)
 
 
@@ -112,7 +112,7 @@ def test_normalize_or_id_invalid_noid_length_too_long():
     or_id = "or-a1b2c3da1b2c3d"
     # Act and Assert
     with raises(ValueError) as excinfo:
-        normalized_or_id = normalize_or_id(or_id)
+        _ = normalize_or_id(or_id)
     assert "Invalid noid length" in str(excinfo.value)
 
 
@@ -121,7 +121,7 @@ def test_normalize_or_id_invalid_no_hyphen_seperator():
     or_id = "or_a1b2c3d"
     # Act and Assert
     with raises(ValueError) as excinfo:
-        normalized_or_id = normalize_or_id(or_id)
+        _ = normalize_or_id(or_id)
         assert "Could not split" in str(excinfo.value)
 
 
@@ -143,7 +143,7 @@ def test_is_event_valid_with_invalid_event():
     # Arrange
     event = json.loads(S3_MOCK_INVALID_EVENT)
     # Act and Assert
-    with raises(InvalidEventException) as error:
+    with raises(InvalidEventException) as _:
         is_event_valid(event)
 
 
@@ -153,7 +153,7 @@ def test_is_event_valid_missing_field():
     event_dict["Records"][0]["s3"].pop("bucket")
     # Act and Assert
     with raises(InvalidEventException) as excinfo:
-        event_valid = is_event_valid(event_dict)
+        _ = is_event_valid(event_dict)
     assert "Not all fields are present" in str(excinfo.value)
```

File: tests/test_helpers.py (hunk 1 above)
Comment (WardMaes) [reply tests/test_helpers.py]: I removed the variables completely, so the assignment won't cause any confusion. The underscore variable could be used when destructuring, but this is not the case here.

**Review by RudolfDG — APPROVED** [review]

**Review by maartends — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**